packages/ionic/src/commands/package/package.ts

```diff
@@ -0,0 +1,270 @@
+import * as fs from 'fs';
+import * as https from 'https';
+
+import { CommandMetadata } from '../../definitions';
+import { isSuperAgentError } from '../../guards';
+import { Command } from '../../lib/command';
+import { FatalException } from '../../lib/errors';
+
+import { fileUtils } from './utils';
+
+const debug = Debug('ionic:commands:package:build');
+const PLATFORMS = ['android', 'ios'];
+const BUILD_TYPES = ['debug', 'release', 'development', 'ad-hoc', 'app-store', 'enterprise'];
+
+interface PackageBuild {
+  job_id: number;
+  id: string;
+  caller_id: number;
+  platform: string;
+  build_type: string;
+  created: string;
+  finished: string;
+  state: string;
+  commit: any;
+  stack: any;
+  profile_tag: string;
+  automation_id: number;
+  environment_id: number;
+  native_config_id: number;
+  automation_name: string;
+  environment_name: string;
+  native_config_name: string;
+  job: any;
+}
+
+interface DownloadUrl {
+  url: string | null;
+}
+
+export class BuildCommand extends Command {
+  async getMetadata(): Promise<CommandMetadata> {
+    return {
+      name: 'build',
+      type: 'project',
+      summary: `Creates a package build on AppFlow using name parameters.`,
+      inputs: [
+        {
+          name: 'platform',
+          summary: `The platform to package (${PLATFORMS.map(v => chalk.green(v)).join(', ')})`,
+          validators: [validators.required, contains(PLATFORMS, {})],
+        },
+        {
+          name: 'type',
+          summary: `The build type (${BUILD_TYPES.map(v => chalk.green(v)).join(', ')})`,
+          validators: [validators.required, contains(BUILD_TYPES, {})],
+        },
+      ],
+      options: [
+        {
+          name: 'securityProfile',
+          summary: 'Security profile',
+          type: String,
+          spec: { value: 'name' },
+        },
+        {
+          name: 'environment',
+          summary: 'The group of environment variables exposed to your build',
+          type: String,
+          spec: { value: 'name' },
+        },
+        {
+          name: 'nativeConfig',
+          summary: 'The group of native config variables exposed to your build',
+          type: String,
+          spec: { value: 'name' },
+        },
+        {
+          name: 'commit',
+          summary: 'Commit (defaults to HEAD)',
+          type: String,
+          groups: [OptionGroup.Advanced],
+          spec: { value: 'sha1' },
+        },
+        {
+          name: 'targetPlatform',
+          summary: 'Target platform',
+          type: String,
+          groups: [OptionGroup.Advanced],
+          spec: { value: 'name' },
+        },
+        {
+          name: 'buildFileName',
+          summary: 'The name for the downloaded build file',
+          type: String,
+          groups: [OptionGroup.Advanced],
+          spec: { value: 'name' },
+        },
+      ],
+    };
+  }
+
+  async run(inputs: CommandLineInputs, options: CommandLineOptions): Promise<void> {
```

Comment:
When I run the command without inputs, I get this:

```
❯ ionic package build
[ERROR] platform must not be empty.
[ERROR] platform must be one of: android, ios (not unset)
[ERROR] type must not be empty.
[ERROR] type must be one of: debug, release, development, ad-hoc, app-store, enterprise (not unset)
Use the --help flag for more details.
```

If you want to, you can prompt for these values in a `preRun` method. A good example is in `packages/ionic/src/commands/init.ts`.

Reply:
this is pretty useful: adding it